scene/gui/tab_container.cpp

```diff
@@ -808,11 +828,16 @@ Size2 TabContainer::get_minimum_size() const {
 }
 
 void TabContainer::set_popup(Node *p_popup) {
-	bool had_popup = get_popup();
-
 	Popup *popup = Object::cast_to<Popup>(p_popup);
-	popup_obj_id = popup ? popup->get_instance_id() : ObjectID();
 
+	ObjectID popup_id = popup ? popup->get_instance_id() : ObjectID();
+	if (popup_id == popup_obj_id) {
+		return;
+	}
+
+	popup_obj_id = popup_id;
+
+	bool had_popup = get_popup();
```

Comment:
This changes the logic, `get_popup()` returns the value based on `popup_obj_id` which you just overwritten.